src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -47,21 +53,18 @@ public ModelStateDictionary(int maxAllowedErrors)
         /// </summary>
         /// <param name="dictionary">The <see cref="ModelStateDictionary"/> to copy values from.</param>
         public ModelStateDictionary(ModelStateDictionary dictionary)
+            : this(dictionary?.MaxAllowedErrors ?? DefaultMaxAllowedErrors)
```

Comment:
This seems like a simpler solution to doing all this. It doesn't null-ref and throws the ArgNull.